grype/db/v6/affected_package_store.go

```diff
@@ -112,15 +170,34 @@ func (s *affectedPackageStore) getNonDistroPackageByName(packageName string, con
 }
 
 func (s *affectedPackageStore) getPackageByNameAndDistro(packageName string, config GetAffectedPackageOptions) ([]AffectedPackageHandle, error) {
+	var resolvedDistros []OperatingSystem
+	var err error
+	if config.Distro != NoDistroSpecified || config.Distro != AnyDistroSpecified {
+		resolvedDistros, err = s.resolveDistro(*config.Distro)
+		if err != nil {
+			return nil, fmt.Errorf("unable to resolve distro: %w", err)
+		}
+
+		if len(resolvedDistros) == 0 {
+			return nil, ErrDistroNotPresent
+		}
```

Comment:
Isn't this unnecessary now as it is covered by the switch statement that follows?